### PR TITLE
MGMT-8705: Collect capi-provider-agent logs

### PR DESCRIPTION
**Pull request by eranco74**
Assisted-test-infra can run cluster-api-provider-agent e2e tests.
This PR adds log gathering of the provider pod and the relevant CRs content

### CODE REVIEW

**Review by avishayt — APPROVED** [review]

**Comment by eranco74** [comment]
/test e2e-metal-assisted-capi
/test e2e-metal-assisted-kube-api-late-binding-single-node
Let's see if we get the cluster logs

**Comment by osherdp** [comment]
> /test e2e-metal-assisted-capi /test e2e-metal-assisted-kube-api-late-binding-single-node Let's see if we get the cluster logs

when running optional jobs, always hold
otherwise, it will just continue with merging it without waiting for those jobs
/hold

File: scripts/download_logs.sh [hunk 1]
```diff
@@ -58,5 +59,25 @@ function collect_kube_api_resources() {
...
+    ${KUBECTL} get pods -n ${NAMESPACE} -o=custom-columns=NAME:.metadata.name --no-headers | xargs -r -I {} sh -c "${KUBECTL} logs {} -n ${NAMESPACE} --all-containers > ${LOGS_DEST}/k8s_{}.log" || true
+    skipper run ./src/junit_log_parser.py --src "${LOGS_DEST}" --dst "${JUNIT_REPORT_DIR}"
+}
+
+function get_pod_namespace() {
+  ${KUBECTL} get pods --no-headers -A  -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.metadata.namespace}{"\n"}' | grep $1 | awk -F " " '{print $2}'
+}
+
+function collect_capi_provider_resources() {
+    JSONPATH=
```
Comment (osherdp) [inline scripts/download_logs.sh:74]: ?

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
can you add to PR description this resources you're gathering?

File: scripts/download_logs.sh [hunk 2]
```diff
@@ -58,5 +59,25 @@ function collect_kube_api_resources() {
...
+    for CR in "${CAPI_PROVIDER_CRS[@]}"
+    do
+      ${KUBECTL} get ${CR} --all-namespaces -o jsonpath='{range .items[*]}{@.metadata.name}{" -n "}{@.metadata.namespace}{"\n"}{end}' --no-headers |\
```
Comment (osherdp) [inline scripts/download_logs.sh:77]: what does this line do?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: scripts/download_logs.sh (hunk 2 above)
Comment (eranco74) [reply scripts/download_logs.sh:77]: Get the CR from all namespaces.
Unlike the assisted-service CRs (though I'm pretty sure that's not entirely true) the provider CRs get created on different namespaces.
This line get all relevant CRs name and namespaces and pass it on the the `kubectl get -o json ...` command

Comment (eranco74) [reply scripts/download_logs.sh:77]: Added a comment in the code

Comment (osherdp) [reply scripts/download_logs.sh:77]: hmm, but the code is still unreadable I'm afraid
can you make it in a for loop?

**Review by osherdp — CHANGES_REQUESTED** [review]

**Review by osherdp — APPROVED** [review]

File: scripts/download_logs.sh (hunk 2 above)
Comment (eranco74) [reply scripts/download_logs.sh]: Done,
Also updated the existing `collect_kube_api_resources` to use for loop

**Pull request merged**